pkg/asset/store/assetsfetcher.go

```diff
@@ -0,0 +1,64 @@
+package store
+
+import (
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Engine used to retrieve and resolve a specified set of assets.
+type AssetsFetcher interface {
+	// Fetchs all the writable assets from the configured assets store.
+	Run(assets []asset.WritableAsset) error
```

Comment:
I kept it from the original implementation, in reality it should be something like `FetchAndPersist`, since it also takes care of persisting the assets

Reply:
renamed